src/core/ext/xds/xds_api.cc

```diff
@@ -972,8 +972,8 @@ std::string XdsApi::ConstructFullResourceName(absl::string_view authority,
                                               absl::string_view resource_type,
                                               absl::string_view name) {
   if (absl::StartsWith(name, "xdstp:")) {
```

Comment:
This should now check the prefix in the authority, not the name.

Reply:
done